gui/wxpython/startup/guiutils.py

```diff
@@ -751,4 +697,16 @@ def switch_mapset_interactively(guiparent, giface, dbase, location, mapset,
             if show_confirmation:
                 GMessage(parent=guiparent,
                          message=_("Current mapset is <%s>.") % mapset)
-            giface.currentMapsetChanged.emit(dbase=None, location=None, mapset=mapset)
+            giface.currentMapsetChanged.emit(dbase=None,
+                                             location=None,
+                                             mapset=mapset)
+
+    if fallback_session:
+        tmp_dbase = os.environ["TMPDIR"]
+        tmp_loc = cfg.temporary_location
+        if tmp_dbase != gisenv()["GISDBASE"]:
+            # Delete temporary location
+            delete_location(tmp_dbase, tmp_loc)
+            giface.grassdbChanged.emit(
```

Comment:
Note here in a comment that we pretend that the whole database for deleted because we assume it is not useful after deleting the temporary location.